deposit/src/main/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJob.java

```diff
@@ -396,40 +409,43 @@ private void ingestWork(ContentContainerObject parent, Resource parentResc, Reso
 		boolean skip = skipResumed(childResc);
 		if (skip) {
 			obj = repository.getWorkObject(childPid);
+			ingestChildren(obj, childResc);
+
+			// Avoid adding primaryObject relation for a resuming deposit if already present
+			if (!obj.getResource().hasProperty(Cdr.primaryObject)) {
+				// Set the primary object for this work if one was specified
+				Statement primaryStmt = childResc.getProperty(Cdr.primaryObject);
```

Comment:
This refactoring looks good!  You may want to consider making a addPrimaryObject helper method in the test that does lines 417-421 so it doesn't have to be duplicated below, but i'll leave that up to your discretion